Bio/phenotype/PhenMicro.py

```diff
@@ -0,0 +1,1126 @@
+
+# Private csv headers - hardcoded because this are supposedly never changed
+_datafile = 'Data File'
+_plate = 'Plate Type'
+_strainType = 'Strain Type'
+_sample = 'Sample Number'
+_strainName = 'Strain Name'
+_strainNumber = 'Strain Number'
+_other = 'Other'
+_hour = 'Hour'
+_file = 'File'
+_position = 'Position'
+_setupTime = 'Setup Time'
+
+_platesPrefix = 'PM'
+_platesPrefixMammalian = 'PM-M'
+#
+
+# Json identifiers - hardcoded as they are set by the creators of opm
+_csvData = 'csv_data'
+_measurements = 'measurements'
+#
+
+
+class PlateRecord(object):
+    """PlateRecord object for storing Phenotype Microarray plates data.
+
+    A PlateRecord stores all the wells of a particular phenotype
+    Microarray plate, along with metadata (if any). The single wells can be
+    accessed calling their id as an index or iterating on the PlateRecord:
+
+    >>> from Bio import phenotype
+    >>> plate = phenotype.read("plate.csv", "pm-csv")
+    >>> well = plate['A05']
+    >>> for well in plate:
+    ...    print("%s" % well.id)
+    A01
+    A02
+    ...
+
+    The plate rows and columns can be queried with an indexing system similar
+    to NumPy and other matrices:
+
+    >>> print(plate[1])
+    PlateRecord('WellRecord['B01'], WellRecord['B02'], WellRecord['B03'], WellRecord['B04']...WellRecord['B12']')
+
+    >>> print(plate[:,1])
+    PlateRecord('WellRecord['A02'], WellRecord['B02'], WellRecord['C02'], WellRecord['D02']...WellRecord['H02']')
+
+    Single WellRecord objects can be accessed using this indexing system:
+
+    >>> print(plate[1,2])
+    WellRecord('(0.0, 11.0), (0.25, 11.0), (0.5, 11.0), (0.75, 11.0), (1.0, 11.0)...(95.75, 11.0)')
+
+    The presence of a particular well can be inspected with the "in" keyword:
+    >>> 'A01' in plate
+    True
+
+    All the wells belonging to a "row" (identified by the first charachter of
+    the well id) in the plate can be obtained:
+
+    >>> for well in plate.get_row('H'):
+    ...     print("%s" % well.id)
+    H01
+    H02
+    H03
+    ...
+
+    All the wells belonging to a "column" (identified by the number of the well)
+    in the plate can be obtained:
+
+    >>> for well in plate.get_column(12):
+    ...     print("%s" % well.id)
+    A01
+    B12
+    C12
+    ...
+
+    Two PlateRecord objects can be compared: if all their wells are equal the
+    two plates are considered equal:
+
+    >>> plate2 = phenotype.read("plate.json", "pm-json")
+    >>> plate == plate2
+    True
+
+    Two PlateRecord object can be summed up or subracted from each other: the
+    the signals of each well will be summed up or subtracted. The id of the
+    left operand will be kept:
+
+    >>> plate3 = plate + plate2
+    >>> print(plate3.id)
+    PM01
+
+    Many Phenotype Microarray plate have a "negative control" well, which can
+    be subtracted to all wells:
+
+    >>> subplate = plate.subtract_control()
+    """
+
+    def __init__(self, plateid, wells=[]):
```

Comment:
You shouldn't use a mutable default argument, would ``wells=()`` work here? Otherwise use ``wells=None`` and set this to a (new) empty list within the ``__init__`` method.

This is one of the problems checked for by the quantifiedcode automated checks, also pylint which calls it W0102, see e.g. http://pylint-messages.wikidot.com/messages:w0102